### DISCUSSION TITLE
Possible to get JSON data from the Places API?

### DISCUSSION THREAD

**Discussion by Rodeveer** [JustFly1984/react-google-maps-api#3183] [Q&A]
I'm working with a client that wants to get some data from the Google Business Reviews (score etc) but I'm struggling to get the data imported into my application.

I'm working with this repo for my Google Maps and since the Place is just an extension of the maps API I thought....

Someone knows how to get the JSON (in this case) from the following URL with this library? https://maps.googleapis.com/maps/api/place/details/json?placeid=[placeID]&key=[apiKey]

Oh FYI. I don't need the maps loaded on that page. I just need the data that is coming from the request! :)

Thanks in advance!

**Comment by akdeb** [comment]
I don't think this API allows this @Rodeveer 
I didn't see any Places usage/extensions in the docs here https://react-google-maps-api-docs.netlify.app/

You might want to go the usual route of fetching with an axios API call like here https://developers.google.com/maps/documentation/places/web-service/details#place-details-example

**Reply by akdeb** [reply]
I take my answer back. If you use this: https://github.com/googlemaps/js-api-loader
You can fetch search results like this
```
import React, { useState, useEffect } from 'react';
import { loadScript } from '@googlemaps/js-api-loader';

const MyMap = () => {
  const [map, setMap] = useState(null);
  const [placesService, setPlacesService] = useState(null);

  useEffect(() => {
    loadScript({
      apiKey: YOUR_API_KEY,
      libraries: ['places']
    }).then(() => {
      const map = new window.google.maps.Map(document.getElementById('map'), {
        center: { lat: 37.7749, lng: -122.4194 },
        zoom: 12
      });
      setMap(map);
      const placesService = new window.google.maps.places.PlacesService(document.createElement('div'));
      setPlacesService(placesService);
    });
  }, [YOUR_API_KEY]);

  const handleSearch = () => {
    placesService.textSearch({ query: 'pizza' }, (results, status) => {
      if (status === 'OK') {
        console.log(results);
      }
    });
  };

  return (
    <>
      <div id="map" style={{ width: '100%', height: '500px' }}></div>
      <button onClick={handleSearch}>Search for pizza</button>
    </>
  );
};

export default MyMap;
```
